### PR TITLE
Adding the ToolStripMenuItem checked state announcement

### LINKED ISSUE #8307: Screen readers cannot announce checked state of checkable ToolStrip/MenuStrip menu items
### .NET version

.NET 8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, but it was working in [MainMenu](https://learn.microsoft.com/dotnet/api/system.windows.forms.mainmenu) control

### Issue description

Initially reported in https://github.com/dotnet/winforms/issues/7770. Screen readers cannot announce checked state of checkable ToolStrip/MenuStrip menu items due to lack of [Toggle pattern](https://learn.microsoft.com/en-us/dotnet/framework/ui-automation/implementing-the-ui-automation-toggle-control-pattern) implementation.

### Steps to reproduce

1. Add MenuStrip to form, initialize it with menu items (for example via "Insert standard items" button)
2. Make items checkable: set `CheckOnClick` property to true or `CheckState` property to `Checked`
3. Start Narrator and run the app
5. Focus on MenuStrip, select checked menu item — Narrator will not announce that item is checked

Sample app:
[Uploading WinFormsMenuStripChecked.zip…]()

### PR DESCRIPTION
**Pull request by v-elnovikova**
Fixes #8307

## Proposed changes

- Added Toggle pattern support for menu items with CheckOnClick set to true and pre-checked menu items
- Added unit tests for ToolStripMenuItemAccessibleObject to verify pattern implementation

## Customer Impact

- Narrator announces checked state of checkable menu item
- Checkable menu item can be checked or unchecked using UI Automation API

## Regression? 

- No

## Risk

- Minimal

### Before

No checked state announcement:
before

No toggle pattern support:
before - no toggle pattern

### After

Checked state announcement:
after

Toggle pattern support: 
after - toggle pattern

## Accessibility testing

- Accessibility Insights for Windows

## Test environment(s) 

- 7.0.0

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8329)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs [hunk 1]
```diff
@@ -52,6 +52,43 @@ propertyID switch
...
                     _ => base.GetPropertyValue(propertyID)
                 };
+
+            public override string DefaultAction
+            {
+                get
+                {
+                    string? defaultAction = Owner.AccessibleDefaultActionDescription;
+                    return defaultAction
+                        ?? (_owningToolStripMenuItem.CheckOnClick ? SR.AccessibleActionCheck : base.DefaultAction);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs:62]:
```suggestion
                    return Owner.AccessibleDefaultActionDescription
                        ?? (_owningToolStripMenuItem.CheckOnClick ? SR.AccessibleActionCheck : base.DefaultAction);
```

**Review by NikitaSemenovAkvelon — APPROVED** [review]
LGTM

**Review by dreddy-work — APPROVED** [review]
LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs [hunk 2]
```diff
@@ -52,6 +52,42 @@ propertyID switch
                     UiaCore.UIA.AcceleratorKeyPropertyId => _owningToolStripMenuItem.GetShortcutText(),
                     _ => base.GetPropertyValue(propertyID)
                 };
+
+            public override string DefaultAction
+            {
+                get
+                {
+                    return Owner.AccessibleDefaultActionDescription
+                        ?? (_owningToolStripMenuItem.CheckOnClick ? SR.AccessibleActionCheck : base.DefaultAction);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs:61]: @v-elnovikova, @dmitrii-drobotov  - why is

**Pull request merged**